pkg/types/gcp/platform.go

```diff
@@ -11,6 +11,22 @@ const (
 	CreateFirewallRulesDisabled CreateFirewallRules = "Disabled"
 )
 
+// DNSZone stores the information common and required to create DNS zones including
+// the project and id/name of the zone.
+type DNSZone struct {
+	// ID Technology Preview.
+	// When the ProjectID is empty, the DNS zone with this ID will be created and managed in the Service
+	// Project (GCP.ProjectID). When ProjectID is provided, the DNS zone will be created and managed
+	// in the requested ProjectID.
```

Comment:
Nit: The logic regarding ProjectID might be better served with the ProjectID parameter below.